ROI fix and improvement

ericpre opened the pull request:
### Description of the change
When a right hand side axis is added to a signal plot, this axes will receive the matplotlib event instead of the left hand side axis. This doesn't play well with the ROI, because by default, matplotlib send event to the last added axis and most of the time, the right hand side axis is added last, and this breaks matplotlib events connection, because we connect the ROI with the matplotlib events of the left hand side axis

### Progress of the PR
- [x] Add support for passing string instead of string of tuple to index `axes_manager`.
- [x] Fix receiving matplotlib event to the left hand side axis by changing respective z-order of the matplotlib axis. 
- [x] Completely remove right_axis instead of making it invisible when closing it.
- [x] add an changelog entry in the `upcoming_changes` folder (see [`upcoming_changes/README.rst`](https://github.com/hyperspy/hyperspy/blob/RELEASE_next_minor/upcoming_changes/README.rst)),
- [x] Check formatting changelog entry in the `readthedocs` doc build of this PR (link in github checks)
- [x] add tests,
- [x] ready for review.

### Minimal example of the bug fix or the new feature
```python
import hyperspy.api as hs

s = hs.datasets.artificial_data.get_core_loss_eels_line_scan_signal(True)
s.remove_background()

roi = hs.roi.SpanROI(left=450, right=600)
# before this PR, the roi can't be changed interactively
roi.add_widget(s, axes='Electron energy loss')

s_ti = s.isig[roi].integrate1D(axis="Electron energy loss")
```

ericpre committed:
Fix _parse_axes of ROI

ericpre committed:
Fix receiving matplotlib events when an additional ax is added

ericpre committed:
Add entries to changelog

ericpre committed:
Improve docstring

jlaehne commented on hyperspy/drawing/signal1d.py:
```diff
@@ -91,6 +91,9 @@ def create_right_axis(self, color='black', adjust_layout=True):
             self.right_ax.hspy_fig = self
             self.right_ax.yaxis.set_animated(self.figure.canvas.supports_blit)
             self.right_ax.tick_params(axis='y', labelcolor=color)
+            # Needs to set the zorder the ax to get the mouse event for ax
```
'for the ax' or 'of the ax'

jlaehne commented on hyperspy/roi.py:
```diff
@@ -209,13 +209,14 @@ def _parse_axes(self, axes, axes_manager):
-              fits. If not, it will try the signal space.
+                - DataAxis. These will not be checked with signal.axes_manager.
+                - anything that will index signal.axes_manager
+            * Anything that can index signal.axes_manager
```
Maybe have this entry before the 'a tuple of'. Otherwise, it is weird that it says the same as in the line above. Why once 'will index' and once 'can index'?

jlaehne commented on hyperspy/roi.py:
```diff
@@ -209,13 +209,14 @@ def _parse_axes(self, axes, axes_manager):
-            * For any other value, it will check whether the navigation
-              space can fit the right number of axis, and use that if it
-              fits. If not, it will try the signal space.
+                - DataAxis. These will not be checked with signal.axes_manager.
```
',' after `DataAxis` and no dot at the end of the line to be consistent with next line.

jlaehne commented on hyperspy/roi.py:
```diff
@@ -426,18 +429,19 @@ def add_widget(self, signal, axes=None, widget=None, color='green',
 
             * a tuple of:
 
                 - DataAxis. These will not be checked with signal.axes_manager.
```
as above

jlaehne commented on hyperspy/roi.py:
```diff
@@ -426,18 +429,19 @@ def add_widget(self, signal, axes=None, widget=None, color='green',
-            * For any other value, it will check whether the navigation
-              space can fit the right number of axis, and use that if it
-              fits. If not, it will try the signal space.
+            * Anything that can index signal.axes_manager
```
as above

jlaehne commented on hyperspy/roi.py:
```diff
@@ -426,18 +429,19 @@ def add_widget(self, signal, axes=None, widget=None, color='green',
+            * ``None``, it will check whether the widget can be added to the
+              navigator, i.e. if dimensionality matches, and use it if
+              possible, otherwise it will try the signal space. If none of the
+              two attempts works, an error message will be raised.
```
work

ericpre committed:
Improve docstring and fix typos. Consistently returns tuple instead of list or tuple in `_parse_axes`. Raise ValueError if length of provided list/tuple doesn't match ROI dimensionality.

ericpre committed:
Update baseline image

ericpre commented:
Thanks @jlaehne, I improved the dosctrings and consistency in the `_parse_axes` method.

jlaehne commented:
LGTM

jlaehne committed:
typos in docstrings

jlaehne committed:
typos in docstrings

jlaehne commented:
I corrected a few typos in the docstrings. Also realized that there are three `__call__` functions where the docstring refers to the possibility to pass a tuple to the axis argument. Shouldn't we also add the following line to all of them: `* Anything that can index the provided ``axes_manager``.`

ericpre committed:
Consolidate docstring

ericpre commented:
> I corrected a few typos in the docstrings. Also realized that there are three `__call__` functions where the docstring refers to the possibility to pass a tuple to the axis argument. Shouldn't we also add the following line to all of them: `` * Anything that can index the provided ``axes_manager``. ``

Yes, indeed! I consolidated the docstring of the methods using `_parse_axes`.

jlaehne commented:
> > I corrected a few typos in the docstrings. Also realized that there are three `__call__` functions where the docstring refers to the possibility to pass a tuple to the axis argument. Shouldn't we also add the following line to all of them: `* Anything that can index the provided`axes_manager`. `
> 
> Yes, indeed! I consolidated the docstring of the methods using `_parse_axes`.

There is one more occurence of the axis-docstring on line 999 (`CircleROI` class)

ericpre commented:
> > > I corrected a few typos in the docstrings. Also realized that there are three `__call__` functions where the docstring refers to the possibility to pass a tuple to the axis argument. Shouldn't we also add the following line to all of them: `* Anything that can index the provided`axes_manager`. `
> > 
> > 
> > Yes, indeed! I consolidated the docstring of the methods using `_parse_axes`.
> 
> There is one more occurence of the axis-docstring on line 999 (`CircleROI` class)

Done, I have removed the docstring completely so that it is inherited from the parent class.
